frontend/src/ui/Blocks.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { graphql, useFragment } from "react-relay/hooks";
+import type {
+    Blocks_blocks as QueryResult,
+    Blocks_blocks$key,
+} from "../query-types/Blocks_blocks.graphql";
+
+import { match } from "../util";
+import { unreachable } from "../util/err";
+
+
+type Props = {
+    realm: Blocks_blocks$key;
+};
+type BlockData = QueryResult["blocks"][0];
+
+export const Blocks: React.FC<Props> = ({ realm }) => {
+    const blocks = useFragment(
+        graphql`
+            fragment Blocks_blocks on Realm {
+                blocks {
+                    id
+                    title
+                    __typename
+                    ... on Text { content }
+                }
+            }
+        `,
+        realm,
+    ).blocks;
```

Comment:
I would prefer

```suggestion
    const { blocks } = useFragment(
        graphql`
            fragment Blocks_blocks on Realm {
                blocks {
                    id
                    title
                    __typename
                    ... on Text { content }
                }
            }
        `,
        realm,
    );
```

:shrug: